File: usoRetrofit/app/src/main/java/com/example/usoretrofit/MainActivity.java
package com.example.usoretrofit;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.example.usoretrofit.interfaces.PetAPI;
import com.example.usoretrofit.models.Pet;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class MainActivity extends AppCompatActivity {
    EditText edtId;
    TextView tvName, tvAge, tvColor;
    Button btnSearch;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        edtId = findViewById(R.id.edtId);
        tvName = findViewById(R.id.tvName);
        tvAge = findViewById(R.id.tvAge);
        tvColor = findViewById(R.id.tvColor);
        btnSearch = findViewById(R.id.btnSearch);

        btnSearch.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                    find(edtId.getText().toString());
            }
        });
    }

    private void find(String cod){
        Retrofit retrofit=new Retrofit.Builder().baseUrl("http://192.168.80.17/")
                .addConverterFactory(GsonConverterFactory.create()).build();

        PetAPI petAPI = retrofit.create(PetAPI.class);

        //llamada http
        Call<Pet> call = petAPI.find(cod);
        call.enqueue(new Callback<Pet>() {
            @Override
            public void onResponse(Call<Pet> call, Response<Pet> response) {
                try{
                    if(response.isSuccessful()){
                        Pet pet = response.body();
                        tvName.setText(pet.getName());
                        tvAge.setText(pet.getAge());
                        tvColor.setText(pet.getColor());
                    }
                }catch (Exception ex){
                    Toast.makeText(MainActivity.this, ex.getMessage(), Toast.LENGTH_SHORT).show();
                }

            }

            @Override
            public void onFailure(Call<Pet> call, Throwable t) {
                Toast.makeText(MainActivity.this, t.getMessage(), Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: semana19retrofitAPI/app/src/main/java/com/example/semana19retrofitapi/MainActivity.kt
package com.example.semana19retrofitapi

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.activity_main.*
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MainActivity : AppCompatActivity() {
    var dataList = ArrayList<Cat>()


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //agregando adaptador a recyclerView
        recycler_view.adapter= DataAdapter(dataList, this)
        recycler_view.layoutManager= LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)

        //getData
        getData()
    }
    private fun getData(){
        val call: Call<List<Cat>> = ApiClient.getClient.getCats()
        call.enqueue(object:Callback<List<Cat>>{
            override fun onResponse(call: Call<List<Cat>>, response: Response<List<Cat>>) {
                dataList.addAll(response!!.body()!!)
                recycler_view.adapter?.notifyDataSetChanged()
            }

            override fun onFailure(call: Call<List<Cat>>, t: Throwable) {
               Toast.makeText(applicationContext, "Ha ocurrido un error", Toast.LENGTH_SHORT).show()
            }
        })
    }
}